luigi/task_register.py

```diff
@@ -16,6 +16,13 @@
 #
 """
 Define the centralized register of all :class:`~luigi.task.Task` classes.
+
+**DEPRECATION WARNING**
+-----------------------
+
+Since after Luigi 2.4.0, everything in this module, including the exception
+types are now **deprecated** in the sense that it's not public API anymore.
+This is an internal module to luigi.
```

Comment:
Actually I think this is too strict. I think too many rely on this. Though some things could be made `_hidden` to make the API more sensible.